packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java

```diff
@@ -128,6 +130,16 @@ public void onAuthenticationError(int errorCode, CharSequence errString) {
     if (activityPaused && isAuthSticky) {
       return;
     }
+    if (errorCode == BiometricPrompt.ERROR_LOCKOUT) {
```

Comment:
Nice! There are two problems:

- What if activityPaused is true but errorCode is ERROR_LOCKOUT? Need to handle these errorCode's properly.
- stop(false) will attempt to send a `completionHandler.onFailure()`. We should not call `onError()` and then `onFailure()`.

Here's how we can fix these:

- Remove logic from `stop` for calling completionHandler (also remove the "success" input parameter) and add `onSuccess` and `onFailure` calls to appropriate places in code.
- Change the handler in onAuthenticationError() to the following:

```java
switch(errorCode) {
  case ERROR_LOCKOUT:
    completionHandler.onError(...);
    break;
  case ERROR_LOCKOUT_PERMANENT:
    completionHandler.onError(...);
    break;
  case USER_CANCELLED: // If I remember correctly, this is the code that is sent if the user navigates away.
    if (activityPaused && isAuthSticky) {
      return;
    }
    break;
  default:
}
stop();
```

Reply:
Done! It stood out to be ERROR_CANCELED.
Not calling `onFailure()` after `onError` resulted in issue. BiometricPrompt did not be prompt after activity resumes as not calling `onFailure()` resulted in the `authenticateWithBiometrics` being in awaiting state forever.

I have tested the latest commit thoroughly and worked as intended for every scenario I tested.

Reply:
There's something bad going on here..

onError calls `result.error()`
onFailure calls `result.success(false)`

You cannot call these back to back on the same result object. That will cause an exception.

What happens right now if you trigger a lockout? Can you look at the adb logs and let me know?

On Dart, you should see "LOCKOUT" exception but you should also see an exception that tells you should not have called `success(false)`

Reply:
@mehmetf It's not causing any exception. When lockout is trigged _platform_exception with lockout message_ is shown and the biometric prompt closes. There is no other exception(on dart side).

Reply:
Yeah but what about the device side? Did you look at the contents of `adb logcat` ?

Reply:
Here is the logcat .. after 5th incorrect fingerprint. 
https://pastebin.com/vFFM4tqR

Reply:
Thanks!!  This is quite peculiar but since you are not seeing any problem, I guess it is OK.

Reply:
@mehmetf , waiting for it to get pushed to pub. Thank you! 🍻

Reply:
Actually, there's a bug :( 

Turns out the callbacks are called from the executor thread rather than UI thread. I don't know how this was working for you (perhaps you are on an older version of Flutter?). See https://github.com/flutter/plugins/pull/1672 in flight.

Reply:
Oh oh! I forgot that it had to be called on UIThread. I am on 1.5.4. 
Anyway hope it get approved soon and published in pub.